docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/private-api-is-used-on-the-project-level.md

```diff
@@ -128,3 +134,177 @@ class CustomerReader extends SprykerCustomerReader implements CustomerReaderInte
 ```
 
 After the fix re-evaluate the code. The same error shouldn’t be returned.
+
+
+## PrivateApi:Persistence
+
+Repository and entity manager on project level can use the factory but only methods that were declared on project level
+
+### Example of code that causes an upgradability error
+
+`CustomerAccessEntityManager` uses factory method `createCustomerAccessMapper` that was declared on the vendor level.
+
+```php
+use Spryker\Zed\CustomerAccess\Persistence\CustomerAccessEntityManager as SprykerCustomerAccessEntityManager;
+
+/**
+ * @method \Pyz\Zed\Customer\Business\CustomerBusinessFactory getFactory()
+ */
+class CustomerAccessEntityManager extends SprykerCustomerAccessEntityManager implements CustomCustomerAccessEntityManager
+{
+   /**
+     * 
+     */
+    public function setContentTypesToAccessible(CustomerAccessTransfer $customerAccessTransfer): void
+    {
+        $this->getFactory()->createCustomerAccessMapper();
+       ....
+    }
+}
+```
+
+### Example of related error in the Evaluator output
+
+```bash
+------------------------------------------------------------------------------------------------------------------------
+Please avoid usage of Spryker\...\CustomerAccessEntityManager::createCustomerAccessMapper(...) in Pyz/Zed/CustomerAccess/Persistence/CustomerAccessEntityManager
+------------------------------------------------------------------------------------------------------------------------
+```
+
+### Resolving the error
+
+See `Resolving the error` section for the first example on this page.
```

Comment:
Provide the link on all places where you redirect the user of the documentation.

Text should be also changed:

```
Solution is the error is the same as in `Resolving the error`[#link-on-something]
```

Reply:
Changed to `Solution is the error is the same as in [Resolving the error](#privateapi:facade)`